packages/backend-data/src/factory.ts

```diff
@@ -307,14 +308,20 @@ class DataGenerator implements ConstructContainerEntryGenerator {
 
     convertJsResolverDefinition(scope, amplifyApi, schemasJsFunctions);
 
+    const namePrefix = this.name === defaultName ? '' : defaultName;
+
     const ssmEnvironmentEntries =
       ssmEnvironmentEntriesGenerator.generateSsmEnvironmentEntries({
-        [`${this.name}_GRAPHQL_ENDPOINT`]:
+        [`${namePrefix}${this.name}_GRAPHQL_ENDPOINT`]:
           amplifyApi.resources.cfnResources.cfnGraphqlApi.attrGraphQlUrl,
-        [`${this.name}_MODEL_INTROSPECTION_SCHEMA_BUCKET_NAME`]:
+        [`${namePrefix}${this.name}_MODEL_INTROSPECTION_SCHEMA_BUCKET_NAME`]:
           modelIntrospectionSchemaBucket.bucketName,
-        [`${this.name}_MODEL_INTROSPECTION_SCHEMA_KEY`]:
+        [`${namePrefix}${this.name}_MODEL_INTROSPECTION_SCHEMA_KEY`]:
           modelIntrospectionSchemaKey,
+        ['AMPLIFY_DATA_DEFAULT_NAME']: `${namePrefix}${this.name}`,
+        // @deprecated
+        [`${this.name}_GRAPHQL_ENDPOINT`]:
```

Comment:
will this not create a duplicate when `namePrefix` is empty?

Reply:
I was worried about the same thing. Up on [line 311](https://github.com/aws-amplify/amplify-backend/pull/2324/files#diff-ec4062ca129b817f0d020474942b0b354abaebe0d9fe5fa88a370c89797369d8R311), name prefix is left blank if the name is defaulted.

Reply:
If it's not expected, we should avoid adding it. Also wanted to ask what's special about `GRAPHQL_ENDPOINT` that this behavior is not replicated for the other `_MODEL_INTROSPECTION_SCHEMA_BUCKET_NAME` and `_MODEL_INTROSPECTION_SCHEMA_KEY`

Reply:
`GRAPHQL_ENDPOINT` predates the MIS fields and customers may be depending on it directly. Since the other two are newer, @sobolk advised that they can be modified without concern for breaking change.

Also `GRAPHQL_ENDPOINT` is included in the prefixing, however, we are leaving an unprefixed version in place with a @deprecation flag to enable removal with the next major version change.